src/main/java/co/cask/hydrator/plugin/spark/dynamic/ScalaSparkCompute.java

```diff
@@ -388,4 +394,26 @@ public StructuredRecord call(Row row) throws Exception {
       return DataFrames.fromRow(row, schema);
     }
   }
+
+  @Nullable
+  private static Class<?> getDataFrameType() {
+    // For Spark1, it has the DataFrame class
+    // For Spark2, there is no more DataFrame class, and it becomes Dataset<Row>
+    try {
+      return ScalaSparkCompute.class.getClassLoader().loadClass("org.apache.spark.sql.DataFrame");
+    } catch (ClassNotFoundException e) {
+      try {
+        return ScalaSparkCompute.class.getClassLoader().loadClass("org.apache.spark.sql.Dataset");
+      } catch (ClassNotFoundException e1) {
+        LOG.warn("Failed to determine the type of Spark DataFrame. " +
+                   "DataFrame is not supported in the ScalaSparkCompute plugin.");
+        return null;
+      }
+    }
+  }
+
+  private static <T> T invokeDataFrameMethod(Object dataFrame, String methodName) throws Exception {
```

Comment:
this naming can be confusing since `dataFrame` may not be a DataFrame and the method may not invoke DataFrame method

Reply:
In Spark terminology, it is still `DataFrame`. It actually has the `DataFrame` scala type defined in Scala.